### PR TITLE
Feat: Parser and metadata export for achievements

### PR DESCRIPTION
**Pull request by Andrew-Gan**
### AchieveParser.cs
`protected override List<AchieveMetadata> Parse()`
- utilization of new `Exporter` class to export achievement data as a metadata file
- data read from xml include: achievement id, grade, condition value and reward value

### Program.cs
- registered the new `AchieveParser` into the list of metadata exporters

### AchieveMetadata.cs
- utilization of `SelectNode` and `SelectSingleNode` over `Reader` to avoid iterating over unneeded nodes
- `AchieveMetadata` class includes achievement id (int) and list of `AchieveGradeMetadata` (List<>)
- `AchieveGradeMetadata` class includes achievement grade (or level), condition (goal) and reward (presumably AP earned)

### AchieveMetadataStorage.cs
- Storage class contains functions which will be invoked from Player.cs for updating and sending achievement packets
`public static List<int> GetAchieveIds()`
- return list of IDs of all achievements available in metadata
`AchieveMetadata GetAchieve(int id)`
- return the achievement data associated with the ID
`public static long GetCondition(int id, int grade)`
- return the condition (or goal) needed to reach a certain grade level for a certain achievement
`public static int GetNumGrades(int id)`
- return number of grades available for an achievement

### CODE REVIEW

File: GameDataParser/Parsers/AchieveParser.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+using GameDataParser.Crypto.Common;
+using GameDataParser.Files;
+using Maple2Storage.Types.Metadata;
+
+namespace GameDataParser.Parsers
+{
+    class AchieveParser : Exporter<List<AchieveMetadata>>
+    {
+        public AchieveParser(MetadataResources resources) : base(resources, "achieve") { }
+        protected override List<AchieveMetadata> Parse()
```
Comment (AlanMorel) [inline GameDataParser/Parsers/AchieveParser.cs:12]: Can you leave a blank line in between here?

**Review by AlanMorel — CHANGES_REQUESTED** [review]

File: GameDataParser/Parsers/AchieveParser.cs (hunk 1 above)
Comment (Andrew-Gan) [reply GameDataParser/Parsers/AchieveParser.cs:12]: you mean between 11 and 12 right? okie

File: MapleServer2/Data/Static/AchieveMetadataStorage.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+namespace MapleServer2.Data.Static
+{
+    public static class AchieveMetadataStorage
+    {
+        private static readonly Dictionary<int, AchieveMetadata> achieves = new Dictionary<int, AchieveMetadata>();
+
+        static AchieveMetadataStorage()
+        {
+            using FileStream stream = File.OpenRead("Maple2Storage/Resources/ms2-achieve-metadata");
```
Comment (angelotadeucci) [inline MapleServer2/Data/Static/AchieveMetadataStorage.cs:15]: Use Paths
Example:
`$"{Paths.RESOURCES}/ms2-guild-metadata"`

**Review by angelotadeucci — CHANGES_REQUESTED** [review]

File: GameDataParser/Parsers/AchieveParser.cs (hunk 1 above)
Comment (AlanMorel) [reply GameDataParser/Parsers/AchieveParser.cs:12]: Yeah

**Review by angelotadeucci — APPROVED** [review]

**Review by AlanMorel — APPROVED** [review]

**Pull request merged**